### PR TITLE
Merge constants.js and websocket.sub.js

### LINKED ISSUE #28264: Merge websockets/constants.js and websockets/websocket.sub.js
As discussed  earlier (https://github.com/web-platform-tests/rfcs/pull/67#pullrequestreview-524740045)

We need to merge constants.js and websockets.js

It will make tests more consistent, remove some duplicated code, and make more tests testable with http2.

I will try to work on a patch.

### PR DESCRIPTION
**Pull request by arenevier**
Right now, we have tests using constants.js and websocket.sub.js

Instead, we merge all the functionality into one unique file
constants.sub.js

The test don't have to specify if the Websocket whether secure anymore.
Instead, we run the 3 variants (unsecure, secure and http2) whenever
possible.

Since some tests were duplicated between a Secure and regular version,
we can now get rid of one those two versions.

We also remove Create-Secure-verify-url-set-non-default-port.any.js and
Create-verify-url-set-non-default-port.any.js which seems to have been
obsolete for some time.

Fixes #28264

### CODE REVIEW

**Comment by arenevier** [comment]
This is a large commit, but hopefully, the changes are relatively straightforward. I can split it into smaller parts if needed.

**Comment by arenevier** [comment]
@zcorpan @ricea

File: websockets/constants.sub.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+//This file requires server-side substitutions and must be included as constants.sub.js
```
Comment (zcorpan) [inline websockets/constants.sub.js:1]: This comment can be removed

**Review by zcorpan — CHANGES_REQUESTED** [review]
Nice work!

A few tests don't include ?wss variant, and a few don't include ?wpt_flags=h2 variant. It's not immediately obvious if this is an oversight or intentional for each such test, so it would be nice with a comment that explains why variants are omitted (or at least say that they are intentionally omitted).

**Comment by arenevier** [comment]
> Nice work!
> 
> A few tests don't include ?wss variant, and a few don't include ?wpt_flags=h2 variant. It's not immediately obvious if this is an oversight or intentional for each such test, so it would be nice with a comment that explains why variants are omitted (or at least say that they are intentionally omitted).

There are 3 cases:
-  basic-auth is not relevant in HTTP/2. I have added a comment
-  some tests which check websocket construction params will be literally the same between all variants. But in the latest patch, I still added the variants to them, just to be consisent.
-  websocket streams. I didn't had the variants because I am not familiar with WebSocketStream. Let me know if you think I should add the variants.

**Comment by zcorpan** [comment]
@ricea can you comment re WebSocketStream?

File: websockets/basic-auth.any.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
 // META: global=window,worker
-// META: script=websocket.sub.js
+// META: script=constants.sub.js
+// META: variant=?wss
+//
+// XXX: this test is not relevant for HTTP/2
```
Comment (ricea) [inline websockets/basic-auth.any.js:5]: Why is this not relevant to HTTP/2?

**Review by ricea — APPROVED** [review]
This is a wonderful improvement in coverage and maintainability! Thank you!

**Comment by arenevier** [comment]
> > @ricea can you comment re WebSocketStream?
> 
> It would be good to add the variants for WebSocketStream too. The handshake works the same as the existing WebSocket API.

I have added h2 variants. Note that they are failing with latest chrome (even with `--enable-experimental-web-platform-features` flag).

File: websockets/basic-auth.any.js (hunk 2 above)
Comment (arenevier) [reply websockets/basic-auth.any.js]: Actually, they are relevant. I needed to modify the handler, but now this test is working as expected.

**Comment by ricea** [comment]
> > It would be good to add the variants for WebSocketStream too. The handshake works the same as the existing WebSocket API.
> 
> I have added h2 variants. Note that they are failing with latest chrome (even with `--enable-experimental-web-platform-features` flag).

Thanks. Odd that it is failing. I will take a look, but you can land this without worry about that.

**Comment by arenevier** [comment]
@zcorpan : does the patch look good to you now? If so, can you remove the "changes requested" flag?

**Review by zcorpan — APPROVED** [review]

**Comment by zcorpan** [comment]
Some CI checks failed. I haven't looked into why. @foolip can you check (or delegate)?

We could close and reopen to see if that fixes it, though I want to allow for analysis of what went wrong in case there's a bug somewhere.

**Comment by foolip** [comment]
It looks like the failures now are actually Python lint failures, and running `flake8` locally should be able to reproduce this. Since Python code is modified here, I would guess that's the issue.

**Comment by arenevier** [comment]
There was a flake8 lint error. I have fixed it. Let's see if it fixes the error.

**Comment by arenevier** [comment]
wpt-chrome-dev-stability is failing. A lot of http2 tests fail. I assume it's just because the release version of chrome does not enable websockets by default. @ricea : can you confirm it's the case? If so, is it ok to push?

**Comment by zcorpan** [comment]
> [taskcluster:error] Task timeout after 7200 seconds. Force killing container.
https://community-tc.services.mozilla.com/tasks/CHZB_ZhiTh-KgKbSPKv5mA/runs/0/logs/public/logs/live.log#L46184

It looks like this is caused by too many tests being changed in this PR and many `h2` tests get a TIMEOUT in Chrome, so when the stability check tries to run each test 10 times to verify the test results are stable, it reaches the taskcluster time limit. It's not ideal, but doesn't need to block merging this PR imo.

**Comment by ricea** [comment]
> wpt-chrome-dev-stability is failing. A lot of http2 tests fail. I assume it's just because the release version of chrome does not enable websockets by default. @ricea : can you confirm it's the case? If so, is it ok to push?

Yes, the tests are running on version 91.0.4449.6 but the first version of Chrome with WS/H2 enabled by default is 91.0.4456.0 (the third part of the version number increases monotonically). So as you suspected, the tests are running without WS/H2 enabled.

I'm slightly concerned that these tests are timing out. I would expect either a fallback to regular WebSocket or a quick failure. However, it shouldn't block this PR unless the infra problem is insurmountable.

**Comment by arenevier** [comment]
> I'm slightly concerned that these tests are timing out. I would expect either a fallback to regular WebSocket or a quick failure. However, it shouldn't block this PR unless the infra problem is insurmountable.

For example, in that test: testOpen.done() will never be called if the socket fails to open. That's why we are seeing the timeouts.

**Comment by arenevier** [comment]
anyway: if it's not blocking, can someone push that patch?

**Comment by zcorpan** [comment]
@web-platform-tests/wpt-core-team FYI, I'll admin-merge this per above.
